src/haiway/files/state.py

```
@@ -0,0 +1,123 @@
+        encoding: str | None = None,
+        binary: bool = False,
+    ) -> FileContent:
+        """Read file content."""
+        return await ctx.state(cls).reading(
+            path,
+            encoding=encoding,
+            binary=binary,
+        )
+
+    @classmethod
+    async def write_file(
```

Pylint (3.3.7)

[refactor] Too many arguments (6/5)

(R0913)